src/shared_modules/utils/socketDBWrapper.hpp

```diff
@@ -43,8 +45,49 @@ class SocketDBWrapper final
             [&](const char* body, uint32_t bodySize, const char* header, uint32_t headerSize)
             {
                 std::unique_lock<std::mutex> lock {m_mutex};
-                m_response = std::string(body, bodySize);
-                m_conditionVariable.notify_one();
+                std::string responsePacket(body, bodySize);
+
+                if (responsePacket.empty())
+                {
+                    m_exceptionStr = "Empty DB response";
+                    m_conditionVariable.notify_one();
+                }
+                else if (0 == responsePacket.compare(0, 3, DB_WRAPPER_ERROR))
+                {
+                    m_exceptionStr = "DB query error: " + responsePacket.substr(4);
+                    m_conditionVariable.notify_one();
+                }
+                else if (0 == responsePacket.compare(0, 3, DB_WRAPPER_IGNORE))
+                {
+                    m_exceptionStr = "DB query ignored: " + responsePacket.substr(4);
+                    m_conditionVariable.notify_one();
+                }
+                else if (0 == responsePacket.compare(0, 3, DB_WRAPPER_UNKNOWN))
+                {
+                    m_exceptionStr = "DB query unknown response: " + responsePacket.substr(4);
+                    m_conditionVariable.notify_one();
+                }
+                else if (0 == responsePacket.compare(0, 3, DB_WRAPPER_DUE))
+                {
+                    m_responsePartial.push_back(nlohmann::json::parse(responsePacket.substr(4)));
```

Comment:
The exceptions that `nlohmann::json::parse()` could throw aren't being handled.

Reply:
fixed